packages/fxa-shared/test/metrics/amplitude.js

```diff
@@ -644,6 +685,82 @@ describe('metrics/amplitude:', () => {
       }
     });
 
+    it('errors - if event_type is `fxa_pay_setup - *` and country_iso is more than 2 characters', () => {
+      const event = {
+        ...minimumEvent,
+        event_type: 'fxa_pay_setup - view',
+        language: 'en',
+        country_iso: 'USA',
+      };
+      try {
+        amplitude.validate(event);
+        assert.fail('Validate is expected to fail');
+      } catch (err) {
+        assert.isTrue(err instanceof Error);
+        assert.equal(
+          err.message,
+          `Invalid data: event/country_iso must match pattern "^[A-Z]{2}$", event must match "then" schema`
+        );
+      }
+    });
+
+    it('errors - if event_type is `fxa_pay_subscription_change - *` and country_iso is more than 2 characters', () => {
+      const event = {
+        ...minimumEvent,
+        event_type: 'fxa_pay_subscription_change - view',
+        language: 'en',
+        country_iso: 'USA',
+      };
+      try {
+        amplitude.validate(event);
+        assert.fail('Validate is expected to fail');
+      } catch (err) {
+        assert.isTrue(err instanceof Error);
+        assert.equal(
+          err.message,
+          `Invalid data: event/country_iso must match pattern "^[A-Z]{2}$", event must match "then" schema`
+        );
+      }
+    });
+
+    it('errors - if event_type is `fxa_pay_setup - *` and country_iso less than 2 characters', () => {
+      const event = {
+        ...minimumEvent,
+        event_type: 'fxa_pay_setup - view',
+        language: 'en',
+        country_iso: 'U',
+      };
+      try {
+        amplitude.validate(event);
+        assert.fail('Validate is expected to fail');
+      } catch (err) {
+        assert.isTrue(err instanceof Error);
+        assert.equal(
+          err.message,
+          `Invalid data: event/country_iso must match pattern "^[A-Z]{2}$", event must match "then" schema`
+        );
+      }
+    });
+
+    it('errors - if event_type is `fxa_pay_subscription_change - *` and country_iso is less than 2 characters', () => {
+      const event = {
+        ...minimumEvent,
+        event_type: 'fxa_pay_subscription_change - view',
+        language: 'en',
+        country_iso: 'U',
+      };
+      try {
+        amplitude.validate(event);
+        assert.fail('Validate is expected to fail');
+      } catch (err) {
+        assert.isTrue(err instanceof Error);
+        assert.equal(
+          err.message,
+          `Invalid data: event/country_iso must match pattern "^[A-Z]{2}$", event must match "then" schema`
+        );
+      }
+    });
```

Comment:
I don't think these two test cases are necessary; you've verified it errors already when the country code is invalid with the tests above this (...is more than 2 characters). Now you're really testing the RegExp pattern matching with `ajv` which we can assume already has its own tests.